### PR TITLE
planner: support MaxOneRow clone

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33887

Problem Summary:
1. Support the MaxOneRow clone
2. Add more logs and metrics for apply executor

### What is changed and how it works?
1. Support the MaxOneRow clone
2. Add more logs and metrics for apply executor
```
mysql> show warnings;
+---------+------+---------------------------------------------------+
| Level   | Code | Message                                           |
+---------+------+---------------------------------------------------+
| Warning | 1105 | Some apply executors can not be executed parallel |
+---------+------+---------------------------------------------------+
1 row in set (0.00 sec)

```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
* explain analyze an unparallel apply to see the log and explain analyze result
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/optimizer.go [hunk 1]
```diff
@@ -412,6 +412,9 @@ func enableParallelApply(sctx sessionctx.Context, plan PhysicalPlan) PhysicalPla
 		noOrder := len(apply.GetChildReqProps(outerIdx).SortItems) == 0 // limitation 1
 		_, err := SafeClone(apply.Children()[apply.InnerChildIdx])
 		supportClone := err == nil // limitation 2
+		if err != nil {
```
Comment (winoros) [inline planner/core/optimizer.go:415]: Or we can output as the warn msg.

**Review by qw4990 — APPROVED** [review]

File: planner/core/optimizer.go [hunk 2]
```diff
@@ -414,6 +414,8 @@ func enableParallelApply(sctx sessionctx.Context, plan PhysicalPlan) PhysicalPla
 		supportClone := err == nil // limitation 2
 		if noOrder && supportClone {
 			apply.Concurrency = sctx.GetSessionVars().ExecutorConcurrency
+		} else {
+			sctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("Some apply executors can not be executed parallel"))
```
Comment (taketh) [inline planner/core/optimizer.go:418]: `Some apply operators can not be executed in parallel`

Comment (Reminiscent) [inline planner/core/optimizer.go:418]:
```suggestion
			sctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf("Some apply operators can not be executed in parallel"))
```

Comment (Reminiscent) [reply planner/core/optimizer.go]: Thanks for your review!

**Review by winoros — APPROVED** [review]

**Comment by winoros** [comment]
closed #34065

**Comment by winoros** [comment]
close https://github.com/pingcap/tidb/issues/34065

**Comment by winoros** [comment]
bot trigger test

**Pull request merged**